packages/cdktf-cli/bin/cmds/ui/models/terraform.ts

```diff
@@ -56,6 +56,43 @@ export interface TerraformPlan {
   readonly planFile: string;
 }
 
+
+function filterChangingResources(resources: PlannedResource[]) {
+  const applyActions = [PlannedResourceAction.UPDATE, PlannedResourceAction.CREATE, PlannedResourceAction.DELETE, PlannedResourceAction.READ];
+  return resources.filter(resource => (applyActions.includes(resource.action)));
+}
+export abstract class AbstractTerraformPlan implements TerraformPlan {
```

Comment:
I abstracted the data mangling work and business logic into an abstract class to DRY things up, hope that works for y'all :)

Reply:
Sounds good. Just make sure to consider https://github.com/hashicorp/terraform-cdk/pull/449#discussion_r556809488

Reply:
Yeah good point, I tested this as well :)